enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,207 @@
+change:
+1. Set the `clusteroperator/network` conditions:
+   - Progressing: true
+   - Upgradeable: false
+2. Deploy pods on every node with `restartPolicy: Never` which are responsible
+   for validating the preconditions. If the preconditions are met the pod will
+   exit with code 0. Some of the preconditions that we will check are:
+   - For MTU changes, the new MTU is valid to apply on that node.
+3. Once all the previous pods are finished, deploy other set of pods on every
+   node that will handle the actual change of the MTU. Wait for them to be
+   ready and running.
+4. Cordon every node, we don't want pods created from this point on.
+5. Ensure via machine-config-operator, that upon reboot, configure-ovs will
+   set the br-ex and its physical port MTUs to the new MTU value.
+6. Ensure that the configmap ovnkube-config is synchronized with the new MTU
+   value.
+7. Set the new MTU value to the applied cluster configuration. At this point
+   CNO is commited to the MTU change.
+8. If any step was unsuccesful, the CNO will set the `clusteroperator/network`
+   conditions to:
+   - Progressing: false
+   - Degraded: true
+   Update the operator configuration status with a description of the problem.
+   At this point the process is interrupted and we require manual intervention.
+9. Wait for pods of step 3 to complete.
+10. Force a rollout of the ovnkube-master and ovnkube-node daemonset sequentially
+    and in appropriate order. This will ensure ovn-kubernetes uses the new MTU
+    value as well.
+11. If any of the previous steps failed, reboot the node, wait for the kubelet to
+    be reporting as Ready again and uncordon it. Otherwise just uncordon the node.
+12. Upon completion, set conditions to:
+    - Progressing: false
+    - Degraded: false
+
+The steps to change the MTU performed by pods of previous step 3 are:
+1. So that we don't have nodes doing things at different times and we have
+   everything synchronized, the pods will wait until the MTU value on the
+   applied cluster configuration changes.
+2. Enter the network namespace of every pod in the pod network and change
+   the MTU of `eth0`. At the same time, the pod will change the MTU of the
+   end of the veth pair in the virtual switch.
+3. The pod will change the MTU of the interfaces ovn-k8s-mp0, ovn-k8s-gw0
+   (if in local mode) and br-int sequentially in that order without any delay
+   in between.
+4. The pod will change the MTU of br-ex and it's physical port sequentially
+   in that order without any delay in between. This includes changing the
+   corresponding NM connection profiles.
+5. If any of these steps failed, the pod will exit with code 1, if all were
+   successful it will exit with code 0.
+
+The order of steps as descibed above correspond to a decrease of MTU.
+For an increase of MTU steps 2-4 are inverted, including the order of
+interfaces of steps 3 and 4.
+
+### User Stories
+
+### Configurable MTU for configure-ovs through machine-config
```

Comment:
("User Stories" itself has "`###`", so its sub items need "`####`")

But also, this isn't a user story. User stories talk about things _users_ want to do, but these are all talking about things _we're_ going to do.

The user stories would be things like
- As an administrator, after installing a cluster and realizing that the sdn MTU was set incorrectly, I would like to fix the MTU without needing to reinstall the cluster.
- As an administrator, after upgrading to a faster underlay network, I would like to make the SDN take advantage of the higher underlay MTU.
- As an administrator, after adding new nodes to a cluster on a network with a smaller MTU than the original nodes, I need to lower the MTU of the SDN, to allow communication between the old and new nodes.